### PR TITLE
add GET /api/reviews/:review_id/comments, add error printing middleware

### PR DESCRIPTION
**Pull request by quinnl404**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -102,3 +102,43 @@ describe("GET: /api/reviews", () => {
       });
   });
 });
+
+describe("GET: /api/reviews/:review_id/comments", () => {
+  it("404: when accessing a non-existant review", () => {
+    return request(app).get("/api/reviews/20/comments").expect(404);
+  });
+
+  it("404: when accessing a review with no comments", () => {
```
Comment (RichardNeat) [inline __tests__/app.test.js]: Have a think about whether or not this circumstance should be an error, the review exists and therefore the user input was valid, but it just so happens that review has no comments. Do we want to say that's an error? Or just say here you go - there's no comments for that review.

File: models/reviews.js [hunk 2]
```diff
@@ -36,3 +36,35 @@ exports.fetchReviews = () => {
...
+    )
+    .then(({ rows }) => {
+      const comments = rows;
+      if (!comments.length) {
+        return Promise.reject({
+          status: 404,
+          msg: "No comments found.",
+        });
+      }
+      console.log(comments);
```
Comment (RichardNeat) [inline models/reviews.js]: please remove any console logs before committing :)

File: models/reviews.js [hunk 3]
```diff
@@ -36,3 +36,35 @@ exports.fetchReviews = () => {
...
+      return comments;
+    });
+};
+
+// {
```
Comment (RichardNeat) [inline models/reviews.js]: please remove any commented out code before committing too :)

**Review by RichardNeat — CHANGES_REQUESTED** [review]
Nice one Quinn, really close! Just one of your tests to reconsider regarding a review with no comments. Have another go and then put in another pr once it's done 😄

**Review by alexdevdavis — APPROVED** [review]
Now you're free to merge 👍

**Pull request merged**